File: web_client/src/transport/reconnect_policy.js
export class ReconnectPolicy {
  constructor({ baseMs = 250, maxMs = 5000 } = {}) {
    this.baseMs = baseMs;
    this.maxMs = maxMs;
  }

  delayForAttempt(attempt) {
    return Math.min(this.maxMs, this.baseMs * 2 ** Math.max(0, attempt - 1));
  }
}

File: android_client/app/src/main/java/com/curtcox/terminals/android/connection/ReconnectPolicy.kt
package com.curtcox.terminals.android.connection

import kotlin.math.min

class ReconnectPolicy(
    private val initialDelayMillis: Long = 500,
    private val maxDelayMillis: Long = 15_000,
) {
    fun delayForAttempt(attempt: Int): Long {
        if (attempt <= 0) return 0
        var delay = initialDelayMillis
        repeat(attempt - 1) {
            delay = min(delay * 2, maxDelayMillis)
        }
        return delay
    }
}
